users/views.py

```diff
@@ -1,3 +1,93 @@
-from django.shortcuts import render
+import json, bcrypt, jwt, re
 
-# Create your views here.
+from django.http    import JsonResponse
+from django.views   import View
+
+from .models        import User
+import my_settings
+
+
+class SignUpView(View):
+    
+    def post(self, request):
+        
+        data = json.loads(request.body)
+        MINIMUM_PASSWORD_LENGTH = 8
+        MINIMUM_ACCOUNT_LENGTH  = 5
+
+        try:
+            account       = data['account']
+            division      = data['division']
+            password      = data['password']
+            name          = data['name']
+            phone_number  = data['phone_number']
+            email         = data['email']
+            date_of_birth = data['date_of_birth']
+            gender        = data['gender']
+
+            email_check = re.compile('^[a-zA-Z0-9+-_.]+@[a-zA-Z0-9-]+\.[a-zA-Z0-9-.]+$')
+            email_check = email_check.match(email)
+
+            if len(account) < MINIMUM_ACCOUNT_LENGTH:
+                return JsonResponse({'MESSAGE' : 'INVALID_ACCOUNT'}, status = 400)
+
+            if User.objects.filter(account=account).exists():
+                return JsonResponse({'MESSAGE' : 'DUPLICATE_ACCOUNT'}, status = 400)
+
+            if User.objects.filter(phone_number=phone_number).exists():
+                return JsonResponse({'MESSAGE' : 'DUPLICATE_PHONE_NUMBER'}, status = 400)
+
+            if email_check == None:
```

Comment:
```suggestion
            if not email_check:
```

파이썬에서는 if 조건문 평가시, 반환 되는 값 자체가 boolean 일 경우나, Truthy, Falsy 한 값일 경우,  == False == None == 0등으로 평가하지 않고, 그 값을 그대로 사용합니다. 그 자체로도 True or False를 판단할 수 있기 때문입니다. False한 지 평가하실 때는 not 을 활용해보세요

Reply:
네, 값을 대로 사용하는 방법으로 수정했습니다!